internal/driver/allocation_settings.go

```diff
@@ -0,0 +1,118 @@
+package driver
+
+import (
+	"reflect"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// ControllerAllocationSettings contains all allocation settings for the controller.
+// These will be used to determine the correct sizing for the volumes.
+type ControllerAllocationSettings struct {
+	Minimum MinimumAllocationSettings `json:"minimum" ,yaml:"minimum"`
+}
+
+// MinimumAllocationSettings contains the minimum allocation settings for the controller.
+// It contains the default settings.
+type MinimumAllocationSettings struct {
+	Default AllocationSettings `json:"default" ,yaml:"default"`
+}
+
+// AllocationSettings contains a set of quantities for the filesystem and block PVCs.
+type AllocationSettings struct {
+	Filesystem map[string]Quantity `json:"filesystem" ,yaml:"filesystem"`
+	Block      Quantity            `json:"block" ,yaml:"block"`
+}
```

Comment:
We can have only one minimum values set by the command line arguments. So it's simple to remove `ControllerAllocationSettings` and `MinimumAllocationSettings`, and rename `AllocationSettings` to `MinimumAllocationSettings`. In other words, L13-L27 should be the following code:

```
type MinimumAllocationSettings struct {
    FileSystem..
    Block...
}
```

Probably the current implementation takes account of the ConfigMap based configuration which you suggested before. However, it hasn't been decided whether this approach can be acceptable.